qmi/instruments/santec/tsl_570.py

```diff
@@ -0,0 +1,891 @@
+
+        Parameters:
+            dwell: Dwell between sweep steps in range [0, 1000[s.
+        """
+        dec = 1
+        if dwell < 0.0 or dwell >= 1000.0:
+            raise ValueError(f"Sweep dwell {dwell} not in valid range [0-1000[s.")
+
+        self._write_and_check_errors(f":WAV:SWE:DWEL {dwell:.{dec}f}")
+
+    @rpc_method
+    def get_sweep_dwell(self) -> float:
+        """Get dwell between sweep steps.
+
+        Returns:
+            dwell: dwell in range [0, 1000[s.
+        """
+        return self._ask_float(":WAV:SWE:DWEL?")
+
+    @rpc_method
+    def set_sweep_delay(self, delay: float) -> None:
+        """Set _continuous_ sweep delay. Possible delay range is 0-999.9s in 0.1nm steps. It does not take into account
+        delay in one-way sweeps to return to the start frequency.
+
+        Parameters:
+            delay: Delay between sweeps in range [0, 1000[s.
+        """
+        dec = 1
+        if delay < 0.0 or delay >= 1000.0:
+            raise ValueError(f"Sweep delay {delay} not in valid range [0-1000[s.")
+
+        self._write_and_check_errors(f":WAV:SWE:DEL {delay:.{dec}f}")
+
+    @rpc_method
+    def get_sweep_delay(self) -> float:
+        """Get delay between sweeps.
+
+        Returns:
+            delay: delay in range [0, 1000[s.
+        """
+        return self._ask_float(":WAV:SWE:DEL?")
+
+    @rpc_method
+    def set_sweep_cycles(self, cycles: int) -> None:
+        """Set sweep repetition times. Possible number of repetitions are in range [0, 999].
+
+        Parameters:
+            cycles: Integer in range [0, 999].
+        """
+        if cycles not in range(1000):
+            raise ValueError(f"Sweep cycles {cycles} not in valid cycles range [0-999].")
+
+        self._write_and_check_errors(f":WAV:SWE:CYCL {cycles}")
+
+    @rpc_method
+    def get_sweep_cycles(self) -> int:
+        """Get sweep repetition times.
+
+        Returns:
+            cycles: Integer in range [0, 999].
+        """
+        return self._ask_int(":WAV:SWE:CYCL?")
+
+    @rpc_method
+    def set_sweep_state(self, state: int) -> None:
+        """Set sweep state. Possible states are:
+        0 - Stop
+        1 - Start
+
+        Parameters:
+            state: Integer in range [0, 1].
+        """
+        if state not in [0, 1]:
+            raise ValueError(f"Sweep state value {state} not in valid states range [0-1].")
+
+        self._write_and_check_errors(f":WAV:SWE:STAT {state}")
+
+    @rpc_method
+    def start_repeating_sweep(self) -> None:
+        """Start a repeating sweep."""
+        self._write_and_check_errors(f":WAV:SWE:STAT:REP")
+
+    @rpc_method
+    def get_sweep_state(self) -> int:
+        """Get sweep state. Possible states are:
+        0 - Stopped
+        1 - Running
+        3 - Standing by trigger
+        4 - Preparation for sweep start
+
+        Returns:
+            state: Integer in range [0, 4].
+        """
+        return self._ask_int(":WAV:SWE:STAT?")
```

Comment:
can return enum here

Reply:
Yes, but I don't want to make many enums, and at times the set- and get- enums do not match. So it is a bit too much work.